### ISSUE TITLE
Failed to prepare subPath for volumeMount "docker-sock"

### ISSUE THREAD

**Issue by DSchmidtDev** [argoproj/argo#803]
**BUG REPORT**

**What happened**:
```bash
argo submit https://raw.githubusercontent.com/argoproj/argo/master/examples/hello-world.yaml
```
results in `Error: failed to prepare subPath for volumeMount "docker-sock" of container "wait"` on Kubernetes v1.9.6

**What you expected to happen**:
Works as usual

**How to reproduce it (as minimally and precisely as possible)**:
1. Setup a Kubernetes cluster with a version greater 1.9.3
2. Setup argo with `argo install`
3. Execute `argo submit https://raw.githubusercontent.com/argoproj/argo/master/examples/hello-world.yaml`

**Environment**:
- Argo version: v2.0.0

- Kubernetes version : 1.9.6

**Other debugging information (if applicable)**:
- workflow result:
Name:             hello-world-7rfdg
Namespace:        xxx
ServiceAccount:   default
Status:           Running
Created:          Fri Mar 23 10:02:32 +0100 (6 minutes ago)
Started:          Fri Mar 23 10:02:32 +0100 (6 minutes ago)
Duration:         6 minutes 29 seconds

STEP                  PODNAME            DURATION  MESSAGE
 hello-world-7rfdg  hello-world-7rfdg  6m

**Comment by sdtokkolabs** [comment]
I'm having the same issue

## Environment:
- Argo version: v2.0.0
- Kubernetes version : 1.9.6

**Comment by kindermoumoute** [comment]
Might be linked to https://github.com/kubernetes/kubernetes/issues/61076

**Comment by DSchmidtDev** [comment]
Linked to https://github.com/kubernetes/kubernetes/issues/61563
and https://github.com/kubernetes/kubernetes/issues/61377

**Comment by adamresson** [comment]
On GKE, was on 1.8.7 without issues. Upgrade to 1.8.9 and hit this. Upgraded again to 1.9.3 and it went away (the subpath volume fix that broke this is in 1.8.9+ & 1.9.4+)

**Comment by alexmt** [comment]
Thanks to @DSchmidtDev for fixing it!
